### PR TITLE
.codecov.yml: move root so files are tracked

### PR DESCRIPTION
**Pull request by JCount**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Currently, all files appear to be untracked on Codecov. This attempts to rectify this by correctly pointing to the root location of the `.codecov.yml` file and while also adjusting the path to be one directory up, in `Library/Homebrew/`.

### CODE REVIEW

File: Library/Homebrew/test/.codecov.yml [hunk 1]
```diff
@@ -1,7 +0,0 @@
-comment: false
-coverage:
-  fixes:
-    - "::Library/Homebrew/"
-  parsers:
-    v1:
-      include_full_missed_files: yes
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/.codecov.yml]: Do we not need/want this any more?

Comment (JCount) [inline Library/Homebrew/test/.codecov.yml]: That appears to only apply to [v1.0 of Codecov](https://docs.codecov.io/v1.0/docs/language-configuration), and the current version is 4.3.6.

**Comment by reitermarkus** [comment]
Do not merge because this doesn't seem to change anything.

File: Library/Homebrew/test/.codecov.yml (hunk 1 above)
Comment (reitermarkus) [inline Library/Homebrew/test/.codecov.yml]: It says in the docs that this is deprecated. Not sure if this was fixed in some way.

**Comment by JCount** [comment]
I wan't going to, I was just rebasing to remove the first commit.

**Comment by reitermarkus** [comment]
Merging this now, maybe it will work once it is in `master`. Otherwise I'll contact support again.

**Pull request merged**